src/MagnumPlugins/ObjImporter/ObjImporter.cpp

```diff
@@ -28,81 +28,344 @@
 #include <fstream>
 #include <limits>
 #include <sstream>
+#include <cstdlib>
 #include <unordered_map>
 #include <Corrade/Containers/ArrayView.h>
 #include <Corrade/Utility/String.h>
+#include <Corrade/Utility/Directory.h>
 
 #include "Magnum/Mesh.h"
 #include "Magnum/MeshTools/CombineIndexedArrays.h"
 #include "Magnum/MeshTools/Duplicate.h"
+#include "Magnum/Math/Vector3.h"
 #include "Magnum/Math/Color.h"
 #include "Magnum/Trade/MeshData3D.h"
+#include "Magnum/Trade/MeshObjectData3D.h"
+#include "Magnum/Trade/ImageData.h"
+#include "Magnum/Trade/PhongMaterialData.h"
+
+#include "MagnumPlugins/TgaImporter/TgaImporter.h"
+
+using namespace Corrade::Containers;
 
 namespace Magnum { namespace Trade {
 
-struct ObjImporter::File {
-    std::unordered_map<std::string, UnsignedInt> meshesForName;
-    std::vector<std::string> meshNames;
-    std::vector<std::tuple<std::streampos, std::streampos, UnsignedInt, UnsignedInt, UnsignedInt>> meshes;
-    std::unique_ptr<std::istream> in;
+
+/* Not using PhongMaterialData, since we may want to set color and texture to
+ * later decide which flags we set. We do not know whether we have a texture
+ * or color beforehand. */
+struct ObjMaterial {
```

Comment:
Oh, forgot to say: could the internal `Obj*` structs be wrapped in an anonymous namespace as well to prevent symbol leakages? Thank you.